raster/src/main/scala/geotrellis/raster/io/geotiff/ArraySegmentBytes.scala

```diff
@@ -51,12 +51,9 @@ object ArraySegmentBytes {
             val result = Array.ofDim[Array[Byte]](offsets.size)
 
             cfor(0)(_ < offsets.size, _ + 1) { i =>
-              byteReader.position(offsets(i).toInt)
-              result(i) = byteReader.getSignedByteArray(byteCounts(i).toInt)
+              result(i) = byteReader.getSignedByteArray(byteCounts(i), offsets(i))
```

Comment:
This maybe a conversation that already happened but why are there `ByteReaderExtensions` when we control the trait we're extending? Why shouldn't `getSignedByteArray` and others be methods on `ByteReader` trait directly?